flocker/node/agents/ebs.py

```diff
@@ -393,6 +375,10 @@ def _next_device(self, instance_id):
             file_name = u'/dev/sd' + suffix
             if file_name not in devices:
                 return file_name
+
+        # Could not find any suitable device that is available
+        # for attachment. Log to Eliot before giving up.
+        NO_AVAILABLE_DEVICE(devices=devices).write()
```

Comment:
> it'd be even more useful (I think) if it included the blockdevice_id and the real allocated device path...the one found by _wait_for_new_device (above) which is different, right?

Agree. But then, we will have to pass ``blockdevice_id`` to ``_next_device`` just for logging. i am slightly inclined to not do that because this is going to go away once FLOC-1887 is in, as you pointed out. If you feel strongly about it, happy to make the change. Thanks!

> the one found by _wait_for_new_device (above) which is different, right?
``_wait_for_new_device`` is after attach happened, so, there will be a new allocated device path.
At ``_next_device``, we haven't yet attempted an attach.